docs/resources/k8s_cluster_onboarding_credential.md

```diff
@@ -0,0 +1,53 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "illumio-cloudsecure_k8s_cluster_onboarding_credential Resource - illumio-cloudsecure"
+subcategory: ""
+description: |-
+  Manages a credential (client_id/client_secret pair) that can be used to onboard one or more k8s clusters onto CloudSecure within a specific Illumio Region.
+---
+
+# illumio-cloudsecure_k8s_cluster_onboarding_credential (Resource)
+
+Manages a credential (`client_id`/`client_secret` pair) that can be used to onboard one or more k8s clusters onto CloudSecure within a specific Illumio Region.
+
+## Example Usage
+
+```terraform
+resource "illumio-cloudsecure_k8s_cluster_onboarding_credential" "example" {
+  illumio_region = "aws-us-west-2"
+  name           = "Dev clusters AWS US"
+
+  # Optional attributes
+  description = "Credential to onboard EKS dev clusters in AWS US regions"
+}
+
+output "example_client_id" {
+  value       = illumio-cloudsecure_k8s_cluster_onboarding_credential.example.client_id
+  description = "The client_id to use to onboard k8s clusters."
+}
+
+output "example_client_secret" {
+  value       = illumio-cloudsecure_k8s_cluster_onboarding_credential.example.client_secret
+  description = "The client_secret to use to onboard k8s clusters."
+  sensitive   = true
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `illumio_region` (String) Illumio Region where the k8s cluster can be onboarded using this credential. An Illumio Region is a designated cloud region where the CloudSecure k8s operators in onboarded k8s clusters connect after onboarding. Choose the Illumio Region nearest to each cluster to maximize performance and security. Must be one of: `aws-ap-southeast-2`, `aws-eu-west-2`, `aws-us-west-2`.
```

Comment:
Should this be: "... An Illumio Region is a designated cloud region where the CloudSecure k8s operators in onboarded k8s clusters connect after **authenticating**"?

Reply:
My intent here was to describe the 2 steps: first, a cluster onboard itself, second, it connects to the Illumio Region using its specific client_id/client_secret that were returned when onboarding. Yes, the connection to the Illumio Region is indeed in two steps itself (authn, then connection to k8sclustersync service), but that's a detail in a documentation about onboarding credentials IMHO. I don't think that the fact that cluster authn is done by the admin plane is relevant.